account_invoice_supplier_self_invoice/models/res_partner.py

```diff
@@ -0,0 +1,46 @@
+# © 2017 Creu Blanca
+# License AGPL-3.0 or later (https://www.gnuorg/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    self_invoice = fields.Boolean(
+        string="Approves self invoice",
+        default=False,
+        help="When checked, all invoices will generate a self-invoice " "on validation",
```

Comment:
If I'm not wrong, this is not for all invoices. It's for vendor bills only, right?

```suggestion
        help="When checked, all vendor bills will generate by default a self-invoice on confirmation.",
```